manifests/conf.pp

```diff
@@ -0,0 +1,278 @@
+# Internal: Install and configure logrotate defaults file, usually
+#   /etc/logrotate.conf
+#
+# see logrotate::rule for description of options.
+#
+# Examples
+#
+#   logrotate::conf{'/etc/logrotate.conf':}
+#
+define logrotate::conf (
+    $ensure          = 'present',
+    $compress        = 'undef',
+    $compresscmd     = 'undef',
+    $compressext     = 'undef',
+    $compressoptions = 'undef',
+    $copy            = 'undef',
+    $copytruncate    = 'undef',
+    $create          = true,
+    $create_mode     = 'undef',
+    $create_owner    = 'undef',
+    $create_group    = 'undef',
+    $dateext         = 'undef',
+    $dateformat      = 'undef',
+    $delaycompress   = 'undef',
+    $extension       = 'undef',
+    $ifempty         = 'undef',
+    $mail            = 'undef',
+    $mailfirst       = 'undef',
+    $maillast        = 'undef',
+    $maxage          = 'undef',
+    $minsize         = 'undef',
+    $missingok       = 'undef',
+    $olddir          = 'undef',
+    $postrotate      = 'undef',
+    $prerotate       = 'undef',
+    $firstaction     = 'undef',
+    $lastaction      = 'undef',
+    $rotate          = '4',
+    $rotate_every    = 'weekly',
+    $size            = 'undef',
+    $sharedscripts   = 'undef',
+    $shred           = 'undef',
+    $shredcycles     = 'undef',
+    $start           = 'undef',
+    $uncompresscmd   = 'undef'
+) {
+
+#############################################################################
+# SANITY CHECK VALUES
+
+  if $name !~ /^[a-zA-Z0-9\._\/-]+$/ {
+    fail("Logrotate::Rule[${name}]: namevar must be alphanumeric")
+  }
+
+  case $ensure {
+    'present','file': {}
+    'absent': {}
+    default: {
+      fail("Logrotate::Rule[${name}]: invalid ensure value")
```

Comment:
All instances of Logrotate::Rule in the fail messages in this class should be Logrotate::Conf